classes/MyTableLister.php

```diff
@@ -535,22 +559,102 @@ public function resolveSQL($sql, $success, $error)
      * @param string $text
-     * @param bool $escape escape for HTML (ie. tranfer ' " <… to &#39; &quot; &lt;…) ?
+     * @param bool $escape escape for HTML?
      * 
      * @return string
      */
     public function translate($text, $escape = true)
     {
-        //@todo kdy může nastat situace, že TableListerTranslate neexistuje? Proč to nenadefinovat jako GodsDev\mycms\Utils\TableListerTranslate ?
-        if (function_exists('TableListerTranslate')) {
-            $result = TableListerTranslate($text);
-        } else {
-            $result = $text;
+        if (isset($this->TRANSLATION[$text])) {
+            $text = $this->TRANSLATION[$text];
         }
         if ($escape) {
-            $result = Tools::h($result);
+            $text = Tools::h($text);
         }
-        return $result;
+        return $text;
+    }
+
+    // custom methods - meant to be rewritten in the class' children
+    
+    /**
+     * Customize particular field's HTML of current $table
+     * @param string $field
+     * @param string $value
+     * @return boolean - true = method was applied so don't proceed with the default, false = method wasn't applied
+     */
+    public function customInput($field, $value)
+    {
+    }
+
+    /**
+     * Custom HTML to be show after detail's edit form but before action buttons
+     * @return string
+     */
+    public function customRecordDetail($record)
+    {
+        return '';
+    }
+
+    /**
+     * Custom HTML to be show after standard action buttons of the detail's form
+     * @return string
+     */
+    public function customRecordActions($record)
+    {
+        return '';
+    }
+
+    /**
+     * Custom saving of a record
+     * @return boolean - true = method was applied so don't proceed with the default, false = method wasn't applied
+     */
+    public function customSave()
+    {
+    }
+
+    /**
+     * Custom deleting of a record
+     * @return boolean - true = method was applied so don't proceed with the default, false = method wasn't applied
+     */
+    public function customAfterDelete()
+    {
+    }
+
+    /**
+     * Called after the $table listing
+     * @return boolean - true = method was applied so don't proceed with the default, false = method wasn't applied
+     */
+    public function customOperation()
+    {
+        return false;
+    }
+
+    /**
+     * Called to optionally fill the search select
+     * @return void
+     */
+    public function customSearch()
+    {
+    }
+
+    /**
+     * Called to optionally fill conditions to WHERE clause of the SQL statement selecting given table
+     * @return void
+     */
+    public function customCondition()
+    {
+    }
+
+    /**
+     * User-defined manipulating with column value of given table  
+     *  
+     * @param string $column
+     * @param array $row
+     * @return mixed
+     */
+    public function customValue($column, array $row)
+    {
+        return $row[$column];
```

Comment:
navrhuji ošetřit neexistenci return isset($row[$column])?$row[$column]:null;
Vadilo-by, že to vrátí totéž, jako kdyby obsahem $row[$column] bylo null ?

Reply:
Dobrý nápad, upravím.
ad 2) imho nevadilo - přijde mi to jako očekávané chování.